examples/threading-multicore/README.md

```diff
@@ -0,0 +1,14 @@
+# threading multicore
+
+## About
+
+This application demonstrates basic threading on multicore.
```

Comment:
I think this is missing some explanations what to expect. Looking at the output and code, I have no idea what to expect or what this demonstrates. :shrug:

Reply:
btw, I get this output:

```
[INFO] - Hello from mid prio thread ThreadId(1) on core CoreId(0)
[INFO] - Hello from mid prio thread ThreadId(2) on core CoreId(0)
[INFO] - ThreadId(1) running again on core CoreId(0)
[INFO] - Hello from low prio thread ThreadId(0) on core CoreId(0)
[INFO] - Hello from high prio thread ThreadId(3) on core CoreId(1)
[INFO] - Hello from high prio thread ThreadId(4) on core CoreId(1)
[INFO] - ThreadId(2) running again (forever) on core CoreId(0)
```

shouldn't the high prio threads run first?

Reply:
The two high priority threads are pinned to core 1 with affinity masks. The reason why they run so late is because:
1. on start_threading, the scheduler on both cores is triggered
1. the faster one executes the `riot-rs-embassy` init thread, because that's the one with highest priority
1. if that's the scheduler on Core 1, the two high prio threads must wait until its execution finished

To prevent such a scenario in a real-world application, we could allow the user to set the core affinity for the embassy thread has well, so that it can be pinned to another core.

Of course ideally, in the above scenario the riot-rs-embassy thread should be preempted and continued on Core 0. But preventing these kinds of priority inversions is quite complex, and out of scope for this PR IMO.

Concerning this concrete example: agreed that it needs better documentation, and I think it should also be simplified.
The current complexity was mostly for testing purposes.

Reply:
Simplified the example now to just start two threads, one with an core-affinity, that both print the core that they are running on: edeaddb.